stdlib/filename.ml

```diff
@@ -100,7 +100,9 @@ module Unix : SYSDEPS = struct
     && (String.length n < 2 || String.sub n 0 2 <> "./")
     && (String.length n < 3 || String.sub n 0 3 <> "../")
   let check_suffix name suff =
-    String.ends_with ~suffix:suff name
+    String.length name >= String.length suff &&
+    String.sub name (String.length name - String.length suff)
+                    (String.length suff) = suff
```

Comment:
What's wrong with `String.ends_with`? It's probably more efficient than this code.

Reply:
This is another recent change in 4.x that was not picked in Multicore.

Reply:
Good catch, there was a revert on 4.12 but then we didn't back that out rebasing to trunk. 

Fix proposed in #10873